crates/testenv/src/lib.rs

```diff
@@ -29,32 +31,48 @@ pub struct TestEnv {
 impl TestEnv {
     /// Construct a new [`TestEnv`] instance with default configurations.
     pub fn new() -> anyhow::Result<Self> {
-        let bitcoind = match std::env::var_os("BITCOIND_EXE") {
-            Some(bitcoind_path) => electrsd::bitcoind::BitcoinD::new(bitcoind_path),
-            None => {
-                let bitcoind_exe = electrsd::bitcoind::downloaded_exe_path()
-                    .expect(
-                "you need to provide an env var BITCOIND_EXE or specify a bitcoind version feature",
-                );
-                electrsd::bitcoind::BitcoinD::with_conf(
-                    bitcoind_exe,
-                    &electrsd::bitcoind::Conf::default(),
-                )
-            }
-        }?;
-
+        let bitcoind_conf = bitcoind::Conf::default();
         let mut electrsd_conf = electrsd::Conf::default();
         electrsd_conf.http_enabled = true;
-        let electrsd = match std::env::var_os("ELECTRS_EXE") {
-            Some(env_electrs_exe) => {
-                electrsd::ElectrsD::with_conf(env_electrs_exe, &bitcoind, &electrsd_conf)
-            }
+        TestEnv::_new(&bitcoind_conf, &electrsd_conf)
+    }
+
+    /// Construct a new [`TestEnv`] instance optionally providing a custom config.
+    pub fn new_with_config(
+        bitcoind_conf: Option<bitcoind::Conf>,
+        electrsd_conf: Option<electrsd::Conf>,
+    ) -> anyhow::Result<Self> {
+        let bitcoind_conf = bitcoind_conf.unwrap_or_default();
+        let electrsd_conf = match electrsd_conf {
+            Some(conf) => conf,
             None => {
-                let electrs_exe = electrsd::downloaded_exe_path()
-                    .expect("electrs version feature must be enabled");
-                electrsd::ElectrsD::with_conf(electrs_exe, &bitcoind, &electrsd_conf)
+                let mut conf = electrsd::Conf::default();
+                conf.http_enabled = true;
+                conf
             }
-        }?;
+        };
+        TestEnv::_new(&bitcoind_conf, &electrsd_conf)
+    }
+
+    /// Construct a new [`TestEnv`] instance with the given configs.
+    fn _new(
```

Comment:
The intention of this is so that we have our own defaults for the config structures? (Enable http)

If this is the case, I would prefer we have our own config struct and impl default there. I think it would be easier to reason about. 

Also, I'm curious about the intention to enable http on default. Is it because electrsd requires it?

Reply:
I pushed a commit implementing your suggestion. The other option is to upstream the change to `electrsd` crate making `http_enabled` the default unless there's a particular reason they chose to set it `false`.

Reply:
I meant to have a config struct that incorporated both `bitcoind::Conf` and `electrsd::Conf`. This way, we can get rid of the `_new` method.

Reply:
Ah, ok. will update soon